pkg/controller/deployment/deployment_controller.go

```diff
@@ -431,13 +431,14 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 	}
 
 	d := obj.(*extensions.Deployment)
+
 	everything := unversioned.LabelSelector{}
 	if reflect.DeepEqual(d.Spec.Selector, &everything) {
 		dc.eventRecorder.Eventf(d, api.EventTypeWarning, "SelectingAll", "This deployment is selecting all pods. A non-empty selector is required.")
 		return nil
 	}
 
-	if d.Spec.Paused {
+	if d.Spec.Paused || d.DeletionTimestamp != nil {
```

Comment:
Please update this and check for the timestamp before checking for paused. https://github.com/kubernetes/kubernetes/pull/20273 changes paused deployments to do more things than simple staus updates.